cfgov/v1/atomic_elements/organisms.py

```diff
@@ -271,6 +271,14 @@ class RelatedPosts(blocks.StructBlock):
                    'posts can match any one topic tag.')
     )
 
+    ignore_tags = blocks.BooleanBlock(
+        required=False,
+        default=False,
+        label="Ignore topic tags",
+        help_text=('If checked, related items shown will be pulled from the '
+                   'full set of posts, without filtering by topic tag.')
+    )
+
```

Comment:
I know this PR is still a draft, and I'm totally snooping, but because I was just poking around with related posts, this is on the top of my mind. I wonder if we should combine the option to ignore topic tags with the option to match all topic tags to make a single field that controls how topic tags are matched. Maybe a `<select>` with three options, like so (we can refine the labels, but this is the general idea):

- Related posts can match any topic tag assigned to the page
- Related posts must match all topic tags assigned to the page
- Related posts ignore topic tags assigned to the page

We use a similar kind of field in a few other spots in Wagtail (like h[ow to filter and sort topics in filter controls](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/v1/atomic_elements/organisms.py#L764-L778)), so it wouldn't be a new pattern. It would also make it clear that checking both the "match all topic tags" and "ignore topic tags" options are mutually exclusive. And, as an added bonus, it would be a good spot to explain that related posts are pulled in based on the topic tags assigned to the page the module is on, which is the reason I was poking around in the code in the first place (to [update the documentation to explain how the related posts are chosen](https://github.cfpb.gov/CFPB/hubcap/wiki/Related-Posts-(Sidebar)#notes)).

Reply:
Snooping is great, no worries!

I like this idea and am going to modify the PR to try having it work accordingly. As two separate checkboxes the fields don't really work/cooperate intuitively, so this would definitely be a better user experiences.

Reply:
Latest commit modifies both BooleanBlocks to instead use a ChoiceBlock for both fields.

The only issue I see with this, that I am not experienced enough to know how to handle, is that this fundamentally changes how the and_filtering value is set and therefore surely would require some sort of database modification to make sure pages that are using that value right now don't break?

Reply:
Yes, this will require a data migration (for which we have some possibly-out-of-date documentation [here](https://cfpb.github.io/consumerfinance.gov/migrations/#data-migrations)) to modify any existing pages that have the existing field set.

I agree that this is a nicer change that adding an additional field. @csebianlander writing data migration can be a little tricky - I looked through the existing ones in the repo and [this](https://github.com/cfpb/consumerfinance.gov/blob/54dddc061b5f843a575a6887bb412a4033de8365/cfgov/v1/migrations/0243_fcm_body_plain_text.py) might be a close example to what we would need for this, if you want to take a stab at it.